emu_sv/density_matrix_state.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+from collections import Counter
+from typing import Any, Iterable
+import torch
+from emu_base import State, DEVICE_COUNT
+from emu_sv.state_vector import StateVector
+from emu_sv.utils import _index_to_bitstring
+
+dtype = torch.complex128
+
+
+class DensityMatrix(State):
+    """Represents a density matrix in a computational basis."""
+
+    # for the moment no need to check positivity and trace 1
+    def __init__(
+        self,
+        matrix: torch.Tensor,
+        *,
+        gpu: bool = True,
+    ):
+        # NOTE: this accepts also zero matrices.
+
+        device = "cuda" if gpu and DEVICE_COUNT > 0 else "cpu"
+        self.matrix = matrix.to(dtype=dtype, device=device)
```

Comment:
I think you should add a factory method for creating a density matrix for the |00..0> state, which would directly create the underlying matrix on the right device, like the `StateVector.make` method